google_guest_agent/agentcrypto/mtls_mds.go

```diff
@@ -208,13 +217,119 @@ func (j *CredsJob) Interval() (time.Duration, bool) {
 	return MTLSScheduleInterval, true
 }
 
-// ShouldEnable returns true if MDS endpoint for fetching credentials is available on the VM.
-// Used for identifying if we want schedule bootstrapping and enable MDS mTLS credential rotation.
+// ShouldEnable implements scheduler job interface which returns true if job
+// should be scheduled based on previous cached [isEnabled] value.
 func (j *CredsJob) ShouldEnable(ctx context.Context) bool {
+	return j.isEnabled.Load()
+}
+
+// checkUserSettings checks and stores user settings for job enablement and the use
+// of OS Native store.
+func (j *CredsJob) checkUserSettings(ctx context.Context, mds *metadata.Descriptor) {
+	logger.Debugf("Checking user settings for %s", j.ID())
+	j.useNativeStore.Store(j.shouldUseNativeStore(mds))
+	j.isEnabled.Store(j.shouldEnableJob(ctx, mds))
+}
+
+// shouldEnableJob returns true if MDS endpoint for fetching credentials is available on the VM
+// and user has not disabled the bootstrapping via config file or Metadata.
+// Used for identifying if we want schedule bootstrapping and enable MDS mTLS credential rotation.
+func (j *CredsJob) shouldEnableJob(ctx context.Context, mds *metadata.Descriptor) bool {
+	var enable bool
+
+	if cfg.Get().MDS != nil {
+		enable = !cfg.Get().MDS.DisableHTTPSMdsSetup
+		logger.Debugf("Found instance config file attribute for enable credential refrehser set to: %t", enable)
```

Comment:
Nit: Here and below, "refresher" instead of "refrehser"

Reply:
Done